### ISSUE TITLE
Restart from UI or CLI does not work

### ISSUE THREAD

**Issue by rchovan** [home-assistant/frontend#7057]
## Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

## The problem
Restart from UI or CLI does not work since 0.114 version

## Expected behavior
HA should be restarted

## Steps to reproduce
Restart from configuration page , or with ha core restart

## Environment

```

arch | x86_64
-- | --
chassis | desktop
dev | false
docker | true
docker_version | 19.03.13
hassio | true
host_os | Debian GNU/Linux bullseye/sid
installation_type | Home Assistant Supervised
os_name | Linux
os_version | 5.8.0-1-amd64
python_version | 3.8.5
supervisor | 245
timezone | Europe/Bratislava
version | 0.115.0
virtualenv | false

```
- Home Assistant release with the issue: 0.115
- Last working Home Assistant release (if known): 0.113
- Browser and browser version: Chrome, Firefox
- Operating system: Debian/testing amd64

## State of relevant entities

```yaml

```

## Problem-relevant configuration

```yaml

```

## Javascript errors shown in your browser console/inspector

nothing in JS console 
![image](https://user-images.githubusercontent.com/6713920/93632165-26b23500-f9ed-11ea-9f82-35cef16f869b.png)

```txt

```

## Additional information
output from CLI
```
ha core restart --log-level=debug
time="2020-09-18T20:17:01+02:00" level=debug msg="Adding homedir to searchpath" homedir=/root
time="2020-09-18T20:17:01+02:00" level=info msg="No configfile found"
time="2020-09-18T20:17:01+02:00" level=debug msg="Debug flags" apiToken=b6d010628bf18edd3569752dbda16c8d116f0077248e51c6d9e0148d8a26b66e56e8ec0cd4934767cd0c2db5c06e3b394793ecc7eb2333e9 cfgFile= endpoint=supervisor logLevel=debug noProgress=false rawJSON=false
time="2020-09-18T20:17:01+02:00" level=debug msg="core restart" args="[]"
time="2020-09-18T20:17:01+02:00" level=debug msg="[GenerateURI]" base=supervisor command=restart section=core
time="2020-09-18T20:17:01+02:00" level=debug msg="[GenerateURI] Result" uri="http://supervisor/core/restart" url="http://supervisor/core/restart" url(string)="http://supervisor/core/restart"
time="2020-09-18T20:17:01+02:00" level=debug msg=Response body="{\"result\": \"error\", \"message\": null}" fields.time=10.1783ms headers="map[Content-Length:[36] Content-Type:[application/json; charset=utf-8] Date:[Fri, 18 Sep 2020 18:17:01 GMT] Server:[Python/3.8 aiohttp/3.6.2]]" received-at="2020-09-18 20:17:01.58710766 +0200 CEST m=+0.012783746" request="&{POST http://supervisor/core/restart HTTP/1.1 1 1 map[Accept:[application/json] Authorization:[Bearer b6d010628bf18edd3569752dbda16c8d116f0077248e51c6d9e0148d8a26b66e56e8ec0cd4934767cd0c2db5c06e3b394793ecc7eb2333e9] User-Agent:[go-resty/2.3.0 (https://github.com/go-resty/resty)]] <nil> 0x73c0a0 0 [] false supervisor map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000096020}" status="400 Bad Request" statuscode=400
```

supervisor logs
```
20-09-18 18:11:34 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:11:34 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:12:39 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:12:39 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:12:49 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:12:49 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:13:11 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:13:11 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:13:13 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:13:13 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:13:17 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:13:17 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:13:30 ERROR (MainThread) [supervisor.utils] Can't execute stop while a task is in progress
20-09-18 18:13:30 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:14:59 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:14:59 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:15:14 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:15:14 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:15:24 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:15:24 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-18 18:15:41 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-18 18:15:41 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
```

## Full gzipped debug log
https://www.dropbox.com/s/qn1jiccxz2hst8p/ha.log.gz?dl=0

**Comment by rchovan** [comment]
anyone ?

**Comment by rchovan** [comment]
Upgraded to 0.115.5 , still available.

ha core info
```
arch: amd64
audio_input: null
audio_output: null
boot: true
image: homeassistant/qemux86-64-homeassistant
ip_address: 172.30.32.1
last_version: 0.115.5
machine: qemux86-64
port: 8123
ssl: false
version: 0.115.5
version_latest: 0.115.5
wait_boot: 600
watchdog: false
```


ha core --log-level=debug restart
```
time="2020-09-30T10:57:25+02:00" level=debug msg="Adding homedir to searchpath" homedir=/root
time="2020-09-30T10:57:25+02:00" level=info msg="No configfile found"
time="2020-09-30T10:57:25+02:00" level=debug msg="Debug flags" apiToken=850837a87b337f62ebc51383cf2ab9538b059d32f7486108dca157ebf959a9a4d5fb13ca07efceb948ab2d139470e24dd07c9fc71759ca15 cfgFile= endpoint=supervisor logLevel=debug noProgress=false rawJSON=false
time="2020-09-30T10:57:25+02:00" level=debug msg="core restart" args="[]"
time="2020-09-30T10:57:25+02:00" level=debug msg="[GenerateURI]" base=supervisor command=restart section=core
time="2020-09-30T10:57:25+02:00" level=debug msg="[GenerateURI] Result" uri="http://supervisor/core/restart" url="http://supervisor/core/restart" url(string)="http://supervisor/core/restart"
time="2020-09-30T10:57:25+02:00" level=debug msg=Response body="{\"result\": \"error\", \"message\": null}" fields.time=20.87306ms headers="map[Content-Length:[36] Content-Type:[application/json; charset=utf-8] Date:[Wed, 30 Sep 2020 08:57:25 GMT] Server:[Python/3.8 aiohttp/3.6.2]]" received-at="2020-09-30 10:57:25.258953804 +0200 CEST m=+0.022429631" request="&{POST http://supervisor/core/restart HTTP/1.1 1 1 map[Accept:[application/json] Authorization:[Bearer 850837a87b337f62ebc51383cf2ab9538b059d32f7486108dca157ebf959a9a4d5fb13ca07efceb948ab2d139470e24dd07c9fc71759ca15] User-Agent:[go-resty/2.3.0 (https://github.com/go-resty/resty)]] <nil> 0x73c0a0 0 [] false supervisor map[] map[] <nil> map[]   <nil> <nil> <nil> 0xc000028028}" status="400 Bad Request" statuscode=400
Error:
```

ha su info
```
addons:
- description: Play Spotify music on your Home Assistant device
  icon: true
  installed: 0.8.1
  logo: true
  name: Spotify Connect
  repository: a0d7b954
  slug: a0d7b954_spotify
  state: started
  version: 0.8.1
- description: Fully featured VSCode experience, to edit your HA config in the browser, including auto-completion!
  icon: true
  installed: 2.7.0
  logo: true
  name: Visual Studio Code
  repository: a0d7b954
  slug: a0d7b954_vscode
  state: started
  version: 2.7.1
addons_repositories:
- https://github.com/hassio-addons/repository
arch: amd64
channel: stable
debug: true
debug_block: false
diagnostics: null
healthy: true
ip_address: 172.30.32.2
logging: info
supported: false
timezone: Europe/Bratislava
version: "245"
version_latest: "245"
wait_boot: 5
```

ha su logs
```

20-09-30 08:51:58 INFO (MainThread) [supervisor.updater] Fetch update data from https://version.home-assistant.io/stable.json

20-09-30 08:56:19 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-30 08:56:19 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
20-09-30 08:57:25 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-09-30 08:57:25 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
```

**Comment by Raul-7-7** [comment]
Exact same issue.

**Comment by Raul-7-7** [comment]
Anyone reading this, just in case you have the same issue, I finally resolved it! it was my stupid Firewall blocking port 8123 to outside world, once I opened that, rebooted container, and it can not restart HA no issue. 
I have NO idea why this was a problem! but hey if it works, dont ask :)

**Comment by spacegaier** [comment]
@rchovan Does the comment from Raul-7-7 apply to your situation as well? If yes, please close the ticket, otherwise let us know.

**Comment by rchovan** [comment]
@spacegaier Hi, it does not work for me, because I'm running HA on localhost with ngix proxy. 

```
server {
    server_name ha.my.domain;

    if ($host = $server_name) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

    listen 80;
    return 404; # managed by Certbot

}

server {
        listen 443 ssl;
        server_name ha.my.domain;
        ssl_certificate /etc/letsencrypt/live/ha.my.domain/fullchain.pem; # managed by Certbot
        ssl_certificate_key /etc/letsencrypt/live/ha.my.domaint/privkey.pem; # managed by Certbot
        ssl_session_timeout 5m;
        autoindex off;

        location ~ /.well-known/acme-challenge/ {
            allow all;
        }

        location / {
            proxy_buffering off;
            proxy_set_header X-Real-IP $remote_addr;
            proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
            proxy_set_header X-Forwarded-Host $server_name;
            proxy_set_header X-Forwarded-Proto https;
            proxy_set_header Host $host;
            proxy_http_version 1.1;
            proxy_set_header Upgrade $http_upgrade;
            proxy_set_header Connection "Upgrade";
            proxy_pass http://127.0.0.1:8123;
        }
        access_log /var/log/nginx/ha_access.log;
        error_log /var/log/nginx/ha_error.log;
}
```

**Comment by rchovan** [comment]
still issue in 0.116.2
```
20-10-11 20:13:21 INFO (SyncWorker_4) [supervisor.docker.homeassistant] Start homeassistant homeassistant/qemux86-64-homeassistant with version 0.116.2
20-10-11 20:13:21 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
20-10-11 20:15:53 INFO (MainThread) [supervisor.snapshots] Found 21 snapshot files
20-10-11 20:17:15 INFO (MainThread) [supervisor.snapshots] Found 21 snapshot files
20-10-11 20:18:11 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-10-11 20:18:11 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
```

**Comment by rchovan** [comment]
Stripped config to minimum :

```
cat configuration.yaml | grep -v "#"

frontend:
api:

default_config:
config:

http:
  server_host: 127.0.0.1
  server_port: 8123

updater:

discovery:

hassio:

conversation:

history:

logbook:

map:

sun:

sensor: !include_dir_merge_list sensors/

group: !include groups.yaml

automation: !include automations.yaml

script: !include scripts.yaml

logger:
  default: debug
  logs:
    homeassistant.components.zha: info
    zigpy: info
    zigpy_deconz: info
    zigpy_zigate: info
    homeassistant.core: info
    hacs: info
    limitlessled: info
    asyncio: info
    bellows: info
    custom_components.mitemp_bt: info
    homeassistant.components.heos: info
    pyheos: info
    homeassistant.components.telegram_bot: info
    homeassistant.components.rest: debug
    homeassistant.components.rest.sensor: debug

system_health:

mobile_app:

```

no custom components, no hacs 
```
ls -l custom_components/
total 0
```

deleted database, and started , still same error. 

```
20-10-13 11:43:58 INFO (SyncWorker_5) [supervisor.docker.interface] Start homeassistant
20-10-13 11:43:59 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
20-10-13 11:47:30 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress
20-10-13 11:47:30 ERROR (MainThread) [supervisor.utils.log_format] Type of message is not string - expected string or bytes-like object
```

**Comment by ex1tium** [comment]
I'm having the same issue here. Cannot update/restart/stop. I stopped all the addons.

**Comment by rchovan** [comment]
still available in 0.117.1
```
20-10-30 12:52:20 INFO (MainThread) [supervisor.homeassistant.core] Updating Home Assistant to version 0.117.1
20-10-30 12:52:20 INFO (SyncWorker_1) [supervisor.docker.interface] Updating image homeassistant/qemux86-64-homeassistant:0.116.4 to homeassistant/qemux86-64-homeassistant:0.117.1
20-10-30 12:52:20 INFO (SyncWorker_1) [supervisor.docker.interface] Downloading docker image homeassistant/qemux86-64-homeassistant with tag 0.117.1.
20-10-30 12:55:50 INFO (SyncWorker_1) [supervisor.docker.interface] Cleaning homeassistant application
20-10-30 12:55:51 INFO (MainThread) [supervisor.homeassistant.core] Successful started Home Assistant 0.117.1
20-10-30 12:55:51 INFO (SyncWorker_3) [supervisor.docker.interface] Cleanup images: ['homeassistant/qemux86-64-homeassistant:0.116.4']
20-10-30 12:56:09 INFO (SyncWorker_3) [supervisor.docker.interface] Cleanup images: ['homeassistant/qemux86-64-homeassistant:0.116.2']
20-10-30 13:05:02 INFO (MainThread) [supervisor.homeassistant] Update pulse/client.config: /data/tmp/homeassistant_pulse
20-10-30 13:05:07 INFO (SyncWorker_3) [supervisor.docker.homeassistant] Starting Home Assistant homeassistant/qemux86-64-homeassistant with version 0.117.1
20-10-30 13:05:07 INFO (MainThread) [supervisor.homeassistant.core] Wait until Home Assistant is ready
20-10-30 14:04:37 ERROR (MainThread) [supervisor.utils] Can't execute restart while a task is in progress

```

is there a way how to list running tasks in supervisor ? To see which task is running ?

**Comment by FerretMonger** [comment]
Is there a solution to this?
